src/searchee.ts

```diff
@@ -160,3 +172,106 @@ export async function createSearcheeFromPath(
 		length: totalLength,
 	});
 }
+
+export function getKeyMetaInfo(stem: string, isAnime: boolean): string {
+	const resM = stem.match(RES_STRICT_REGEX)?.groups?.res;
+	const res = resM ? `.${resM}` : "";
+	const sourceM = sourceRegexParse(stem);
+	const source = sourceM ? `.${sourceM}` : "";
+	const groupM = stem.match(RELEASE_GROUP_REGEX)?.groups?.group;
+	if (groupM) {
+		return `${res}${source}-${groupM}`.toLowerCase();
+	}
+	if (!isAnime) {
+		return `${res}${source}`.toLowerCase();
+	}
+	const groupAnimeM = stem.match(ANIME_REGEX)?.groups?.group;
+	if (groupAnimeM) {
+		return `${res}${source}-${groupAnimeM}`.toLowerCase();
+	}
+	return `${res}${source}`.toLowerCase();
+}
+
+export function getMovieKey(stem: string): {
+	ensembleTitle: string;
+	keyTitle: string;
+	year: number;
+} | null {
+	const match = stem.match(MOVIE_REGEX);
+	if (!match) return null;
+	const keyTitle = reformatTitleForSearching(match.groups!.title, false)
+		.replace(NON_UNICODE_ALPHANUM_REGEX, "")
+		.toLowerCase();
+	if (!keyTitle.length) return null;
+	const year = extractInt(match.groups!.year);
+	const ensembleTitle = `${match.groups!.title}.${year}`;
+	return { ensembleTitle, keyTitle, year };
+}
+
+export function getSeasonKey(stem: string): {
+	ensembleTitle: string;
+	keyTitle: string;
+	season: string;
+} | null {
+	const match = stem.match(SEASON_REGEX);
+	if (!match) return null;
+	const keyTitle = reformatTitleForSearching(match.groups!.title, false)
+		.replace(YEAR_REGEX, "")
+		.replace(NON_UNICODE_ALPHANUM_REGEX, "")
+		.toLowerCase();
+	if (!keyTitle.length) return null;
+	const season = `S${extractInt(match.groups!.season)}`;
+	const ensembleTitle = `${match.groups!.title}.${season}`;
+	return { ensembleTitle, keyTitle, season };
+}
+
+export function getEpisodeAndKey(stem: string): {
```

Comment:
episode and what?

Reply:
It's hard to name. The key is the unique identifier to use if something matches. I separated the info as ensemble uses this too and parses it differently.

Reply:
the ones above seem to follow a convention - `getSeasonKey` `getMovieKey` etc, so with this one being `getEpisodeAndKey` it seems like there should be another word in between `And` and `Key`. Are these ones just different from each other?

Reply:
I see it now. They were previously named like that before their logic changed.